### PR TITLE
feat: added authentication

### PR DESCRIPTION
**Pull request by KanekarSnehal**
### Overview
---
Added authentication with following features
- user can signup to create an account
- added all validations necessary for signup and login form
- added login with test credentials
- user cannot access profile if not logged in
- added protected and guest routes

### Preview
---
image
image
image

### CODE REVIEW

**Review by nerdyspook — APPROVED** [review]
Well structured code, easy to understand.

File: src/pages/Auth/Signup.jsx [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+          <div class="row">
+            <div class="column-30-pc">
+              <label
+                for="first-name"
+                class="form-label text-bold-weight form-label-required-field"
+              >
+                Last Name
+              </label>
+            </div>
+            <div class="column-70-pc">
```
Comment (ashishnnnnn) [inline src/pages/Auth/Signup.jsx:210]: These class should be changed with className

**Review by ashishnnnnn — APPROVED** [review]
Code Looks fine to merge . :+1:

File: src/pages/Auth/Signup.jsx (hunk 1 above)
Comment (KanekarSnehal) [reply src/pages/Auth/Signup.jsx]: Done

**Pull request merged**